docs/saved_objects/saved_object_access_control_design_proposal.md

```diff
@@ -0,0 +1,216 @@
+* **Workspace admin**: Workspace admin have complete access to workspace configurations and saved objects in the workspace. Workspace creator will become the workspace admin when a workspace is created. Workspace admin most time can be used exchangeably with and workspace owner. Workspace admin
+* **Workspace operator**: workspace operator has permission to view, create and update saved objects in the workspace.
+* **Workspace viewer**: workspace operator only has permission to view the saved objects in the workspace
+
+Please note that, the workspace admin, operator and viewer persona are specific to the workspace. e.g. one workspace operator in one workspace can be an admin in another workspace.
+
+super-admin is a stack level role, it can be configured in the `opensearch_dashboards.yml` config file.
+
+### Discretionary Access Control for Data Access Control
+
+In order to support allowing a resource owner to manage the access to the resource feature, we can employ discretionary access control model. The idea is that, for each OpenSearch Dashboards saved object, we can attach an ACL (access control list) to it to save the principals and permission to operate that saved object. When a user is attempting to perform any operation on that saved object, we will first of all evaluate the ACL and then allow or deny the operation accordingly.
+
+#### Why not using role based access control model
+
+In existing OpenSearch products, all access control follows role/attributes based access control model, where a user can have several roles, while each role has corresponding permissions. The issue with the role based access control model is that, it requires administrator to centrally manage all access policies, which defines what principals can perform what actions on what resources. With admin managed access policies, OpenSearch Dashboards users cannot grant other users permissions to access specific resources, nor can they add other users to new workspaces. Creating workspaces and managing user access to workspaces will become admin responsibility, which isn't a simple collaboration experience.
+
+### Workspace and Saved Objects
+
+Workspace is a newly introduced type of OpenSearch Dashboards data, it is a container which organize the functions and saved objects, so the saved object in a workspace inherit the permissions settings from the workspace. One saved object can potentially in multiple workspaces, thus it inherits the permission settings from all the workspaces it is assigned to.
+
+#### Permissions
+
+Based on the proposed workspace experience, an user may have following type of permissions on OpenSearch Dashboards data:
+
+* read
+* write (read_and_write)
+    * the write permission means the user update the saved object, and can add this saved object to another workspace, or update the ACL of the object so that the object can be shared to others
+
+For workspaces, a user may have following 3 type of permissions:
+
+* management
```

Comment:
@zengyan-amazon It seems `management` is not necessary in practice, it's basically `write` + `library_write`, the `library_write` give the user full access to all the saved objects associated with the workspace. The `write` permission will give a user full access to a workspace object includes the ACLs on the object. To my understanding, this equals to `management`, am I right?